### PR TITLE
implement head in go

### LINKED ISSUE #2: Implement head(1) from scratch in Go
... so that `go run . < in.txt` does the same as `head < in.txt`.

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #2

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package main
```
Comment (Al2Klimov) [inline main.go:3]: What's missing here?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		}
+
+		if len(data) == 0 || counter == 10 {
+			break
+		}
+
+		_, _ = os.Stdout.Write(data)
+		counter++
+	}
+
```
Comment (Al2Klimov) [inline main.go:29]: What's this empty line for?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+func main() {
+	buf := bufio.NewReader(os.Stdin)
+	var counter int
+
+	for {
+		data, dataErr := buf.ReadBytes('\n')
+		if dataErr != nil && dataErr != io.EOF {
+			fmt.Fprintln(os.Stderr, dataErr)
+			os.Exit(2)
```
Comment (Al2Klimov) [inline main.go]: Why exit 2?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		}
+
+		if len(data) == 0 || counter == 10 {
+			break
+		}
+
+		_, _ = os.Stdout.Write(data)
+		counter++
```
Comment (Al2Klimov) [inline main.go:28]: And the usual EOF check?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	var counter int
+
+	for {
+		data, dataErr := buf.ReadBytes('\n')
+		if dataErr != nil && dataErr != io.EOF {
+			fmt.Fprintln(os.Stderr, dataErr)
+			os.Exit(1)
+		}
+
+		if len(data) == 0 || counter == 10 {
```
Comment (Al2Klimov) [inline main.go]: What's `len(data) == 0` for?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Run `go run .` and type:

a
b
c
d
e
f
g
h
i
j

Compare with the behavior of head.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/implement-head-in-go-2`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**in.txt:**
```
one
two
three
four
five
six
seven
eight
nine
ten
eleven
twelve
```

**Case 1:**

**Terminal:** `go run . < in.txt`
**Output:**
```
one
two
three
four
five
six
seven
eight
nine
ten
```
**Terminal:** `head < in.txt`
**Output:**
```
one
two
three
four
five
six
seven
eight
nine
ten
```

**Conclusion:**
Both outputs are equal.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**